Sources/AuthFoundation/OAuth2/OAuth2Client.swift

```diff
@@ -272,7 +272,16 @@ public final class OAuth2Client {
     ///   - type: Type of token to revoke.
     ///   - completion: Completion block to invoke once complete.
     public func revoke(_ token: Token, type: Token.RevokeType, completion: @escaping (Result<Void, OAuth2Error>) -> Void) {
-        let tokenType = type.tokenType
+        if type == .all {
+            revokeAll(token, completion: completion)
+            return
+        }
+        
+        guard let tokenType = type.tokenType else {
```

Comment:
In the case where `type.tokenType == nil` it would also be map to `type == .all `. I think we can just call the             `revokeAll(token, completion: completion)` if the unwrapping fails. However, I was wondering why the first check `if type == .all. ` exists if the optional unwrap also covers it.  

Although if the first is true, we would never get to the optional unwrap.

Reply:
I was trying to avoid force-unwrapping `type.tokenType` since its become optional now. There's no guarantee, in the future, that other types could be introduced _other_ than `.all` which may also not return a `tokenType`. I think I could make the `if` on the first line of this function a bit cleaner though.

Reply:
That sounds good. I see your point here.